tests/ui/lint/lint-ctypes-cstr.rs

```diff
@@ -0,0 +1,16 @@
+#![crate_type = "lib"]
+#![deny(improper_ctypes)]
+
+use std::ffi::{CStr, CString};
+
+extern "C" {
+    fn take_cstr(s: CStr);
+    //~^ ERROR `extern` block uses type `CStr`, which is not FFI-safe
+    //~| HELP consider using `*const std::ffi::c_char` instead, and pass the value from `CStr::as_ptr()`
+    fn take_cstring(s: CString);
+    //~^ ERROR `extern` block uses type `CString`, which is not FFI-safe
+    //~| HELP consider using `*const std::ffi::c_char` instead, and pass the value from `CStr::as_ptr()`
+    fn take_cstring_ref(s: &CString);
+    //~^ ERROR `extern` block uses type `CString`, which is not FFI-safe
+    //~| HELP consider using `*const std::ffi::c_char` instead, and pass the value from `CStr::as_ptr()`
+}
```

Comment:
I think there should be tests for creating an extern fn too:
```rust
extern "C" fn uwu(s: CString) {
     // do things idk
 }
 ```
 (and its derivatives)